lib/core/request.c

```diff
@@ -713,19 +713,25 @@ void h2o_req_log_error(h2o_req_t *req, const char *module, const char *fmt, ...)
 
 #undef INITIAL_BUF_SIZE
 
+#define EMIT_ONE(input_, len_)                                                                                                     \
+    if (input_.len < len_) {                                                                                                       \
+        memcpy(p, input_.base, input_.len);                                                                                        \
+        p += input_.len;                                                                                                           \
+    } else {                                                                                                                       \
+        memcpy(p, input_.base, (len_ - 3));                                                                                        \
+        p += len_ - 3;                                                                                                             \
+        memcpy(p, "...", 3);                                                                                                       \
+        p += 3;                                                                                                                    \
+    }
     /* build prefix */
-    char *pbuf = h2o_mem_alloc_pool(&req->pool, char, sizeof("[] in request::") + 32 + strlen(module)), *p = pbuf;
+    char *pbuf = h2o_mem_alloc_pool(&req->pool, char, sizeof("[] in request::") + 32 + 1 + 64 + strlen(module)), *p = pbuf;
```

Comment:
In the original code, space consumed by `:` after the path is calculated as part of the `sizeof` expression. Please follow that convention.